File: VB/ButtonEditImages/Custom Editor/CustomButtonEdit.vb
Imports DevExpress.XtraEditors
Imports System.ComponentModel

Namespace ButtonEditImages

    <ToolboxItem(True)>
    Public Class CustomButtonEdit
        Inherits ButtonEdit

        Shared Sub New()
            RepositoryItemCustomButtonEdit.RegisterCustomButtonEdit()
        End Sub

        Public Sub New()
        End Sub

        <DesignerSerializationVisibility(DesignerSerializationVisibility.Content)>
        Public Overloads ReadOnly Property Properties As RepositoryItemCustomButtonEdit
            Get
                Return TryCast(MyBase.Properties, RepositoryItemCustomButtonEdit)
            End Get
        End Property

        Public Overrides ReadOnly Property EditorTypeName As String
            Get
                Return RepositoryItemCustomButtonEdit.CustomEditName
            End Get
        End Property
    End Class
End Namespace

File: CS/ButtonEditImages/Custom Editor/CustomButtonEdit.cs
using DevExpress.XtraEditors;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ButtonEditImages {
    [ToolboxItem(true)]
    public class CustomButtonEdit : ButtonEdit {
        static CustomButtonEdit() {
            RepositoryItemCustomButtonEdit.RegisterCustomButtonEdit();
        }

        public CustomButtonEdit() {
        }

        [DesignerSerializationVisibility(DesignerSerializationVisibility.Content)]
        public new RepositoryItemCustomButtonEdit Properties {
            get { return base.Properties as RepositoryItemCustomButtonEdit; }
        }

        public override string EditorTypeName {
            get { return RepositoryItemCustomButtonEdit.CustomEditName; }
        }

    }
}
